http/unit-tests/CurlSListTest.cc

```diff
@@ -0,0 +1,244 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+//
+// You can contact OPeNDAP, Inc. at PO Box 112, Saunderstown, RI. 02874-0112.
+
+#include "config.h"
+
+#include <memory>
+#include <cstring>
+#include <iostream>
+
+#include <curl/curl.h>
+#include <curl/easy.h>
+
+#include "CurlUtils.h"
+
+#include "test_config.h"
+
+// Maybe the common testing code in modules should be moved up one level? jhrg 11/3/22
+#include "modules/common/run_tests_cppunit.h"
+
+using namespace std;
+
+#define prolog std::string("CurlSListTest::").append(__func__).append("() - ")
+
+namespace http {
+
+class CurlSListTest : public CppUnit::TestFixture {
+
+public:
+    string d_data_dir = TEST_DATA_DIR;
+
+    // Called once before everything gets tested
+    CurlSListTest() = default;
+
+    // Called at the end of the test
+    ~CurlSListTest() override = default;
+
+    // Called before each test
+    void setUp() override {
+        DBG( cerr << "\n");
+        DBG( cerr << "#-----------------------------------------------------------------\n");
+        DBG( cerr << "setUp() - BEGIN\n");
+        debug = true;
+        DBG( cerr << "setUp() - END\n");
+    }
+
+    // Called after each test
+    void tearDown() override {
+        DBG( cerr << "\n");
+    }
+
+/*##################################################################################################*/
+/* TESTS BEGIN */
+
+
+    string lorem="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris semper laoreet enim, sed porttitor "
+                 "lacus consectetur sit amet. Mauris ultrices aliquet velit, in mattis dolor rutrum et. Curabitur "
+                 "dolor purus, aliquet nec erat ut, ullamcorper vestibulum sem. Donec mattis pharetra justo, vel "
+                 "egestas mi semper sit amet. Sed scelerisque elementum nunc vitae faucibus. Cras mattis facilisis "
+                 "arcu quis aliquet. Morbi rutrum nunc et mattis laoreet. In finibus ante eu ullamcorper tempor. "
+                 "Curabitur blandit, nisl ut blandit malesuada, arcu sem faucibus eros, sit amet congue velit sem "
+                 "in velit. Etiam ut tellus venenatis, sodales elit vel, finibus augue. Integer consequat purus.";
+
+    vector<string> slist_baselines{
+            "FirstName: Willy",
+            "LastName: Wonka",
+            "kjwhebd: jkhbvkwjqehv ljhcljhbwvcqwx",
+            "lorem: "+lorem
+    };
+
+    curl_slist *load_slist(curl_slist *request_headers) const {
+        auto slist = request_headers;
+        //hdrs = curl::append_http_header(hdrs, "Dum", "Dummer");
```

Comment:
Should this commented line be removed? Sonar may issue a code smell.

Reply:
fixed